Cargo.toml

```diff
@@ -12,28 +13,37 @@ categories = ["text-processing"]
 exclude = ["/.github/*", "/Cargo.lock.msrv"]
 
 [features]
-default = ["unicode", "perf"]
+default = ["unicode", "perf", "std"]
 # Enable #[track_caller] in unit tests.
 track_caller = []
+
+std = ["regex/std"]
+
 perf = ["regex/perf"]
 perf-dfa = ["regex/perf-dfa"]
+perf-onepass = ["regex/perf-onepass"]
+perf-backtrack = ["regex/perf-backtrack"]
 perf-inline = ["regex/perf-inline"]
 perf-literal = ["regex/perf-literal"]
 perf-cache = ["regex/perf-cache"]
+
 unicode = ["regex/unicode"]
 
 [dependencies.regex]
-version = "1.3.8"
+version = "1.9.1"
+default-features = false
+
+[dependencies.bit-set]
+version = "0.5"
 default-features = false
-features = ["std"]
 
 [dependencies]
-bit-set = "0.5"
+hashbrown = "0.14.0"
```

Comment:
Have you considered just using a `BTreeMap` when `HashMap` is unavailable instead of adding a new unconditional dependency? This is what the `regex` crate itself does.

Reply:
Seems `BTreeMap` is actually faster than `HashMap` by around 4% in almost all the included benchmarks on my machine (except for `run_backtrack_limit`, where it is 12% slower). Change made.